src/cromshell/counts/command.py

```diff
@@ -0,0 +1,288 @@
+        )
+
+        http_utils.set_and_check_cromwell_server(
+            config=config, workflow_id=resolved_workflow_id
+        )
+
+        # Get metadata
+        formatted_metadata_parameter = metadata_command.format_metadata_params(
+            list_of_keys=config.METADATA_KEYS_TO_OMIT,
+            exclude_keys=True,
+            expand_subworkflows=not compress_subworkflows,
+        )
+
+        workflow_metadata = metadata_command.get_workflow_metadata(
+            meta_params=formatted_metadata_parameter,
+            api_workflow_id=config.cromwell_api_workflow_id,
+            timeout=config.requests_connect_timeout,
+            verify_certs=config.requests_verify_certs,
+            headers=http_utils.generate_headers(config),
+        )
+
+        if json_summary:
+            print_task_status_summary(workflow_metadata=workflow_metadata)
+        else:
+            pretty_status_counts(
+                workflow_id=resolved_workflow_id,
+                workflow_metadata=workflow_metadata,
+                expand_subworkflows=not compress_subworkflows,
+            )
+
+        DelayedLogMessage.display_log_messages()
+    return 0
+
+
+def pretty_status_counts(
+    workflow_id: str, workflow_metadata: dict, expand_subworkflows: bool
+) -> None:
+    """
+    Prints the workflow summary and calls the task to print formatted workflow status
+
+    :param workflow_id: Hexadecimal identifier of workflow submission
+    :param workflow_metadata: Metadata of the workflow to process
+    :param expand_subworkflows: Boolean, whether or not to print subworkflows
+    :return:
+    """
+    workflow_status = workflow_metadata.get("status")
+    print(
+        colored(
+            workflow_id + "\t" + workflow_status,
+            color=io_utils.TextStatusesColor.COLOR_UNDERLINED["color"],
+            attrs=io_utils.TextStatusesColor.COLOR_UNDERLINED["attrs"],
+        )
+    )
+    print_workflow_status(
+        workflow_metadata=workflow_metadata,
+        indent="\t",
+        expand_sub_workflows=expand_subworkflows,
+    )
+
+
+def print_workflow_status(
+    workflow_metadata: dict, indent: str, expand_sub_workflows: bool
+) -> None:
+    """
+    Runs through each task of a workflow and calls a function to
+    print task status counts
+    :param workflow_metadata: Metadata of the workflow to process
+    :param indent: Indent string given as "\t", used to indent print out
+    :param expand_sub_workflows:  Boolean, whether or not to print subworkflows
+    :return:
+    """
+    calls_metadata = workflow_metadata["calls"]
+    calls = list(calls_metadata.keys())
+
+    for call in calls:  # For each task in given metadata
+        # If task has a key called 'subworkflowMetadata' in its first (zero) element
+        # (shard) and expand_sub_workflow parameter is set to true then rerun this
+        # function on that subworkflow
+        if "subWorkflowMetadata" in calls_metadata[call][0] and expand_sub_workflows:
```

Comment:
Both are needed, the first condition makes sure there is a subworkflow present if not then python would fail saying the subworkflow key is not available. The second condition makes sure to only look through subworkflows only if the user wants to see them.

is there another approach to this?

Reply:
I'm not understanding the following part:
> python would fail saying the subworkflow key is not available

Reply:
When a workflow has a subworkflow, the call in the main workflow for the subworkflow is represented by "ScatterAt#_#" in the metadata. Within that subworkflow is a set of dictionaries (could be 1 or more depending on the number of times the subworkflow is executed). Within the dictionaries is a key called "subWorkflowMetadata", the value for this key is a dictionary holding the metadata for the subworkflow. 
In the 'for' block below we use the "subWorkflowMetadata" dictionary value as input to the print_workflow_status function, if that key is not present it will fail.

If the `if "subWorkflowMetadata" in calls_metadata[call][0]` wasn't there and we come across a none-subworkflow call in the main workflow. That call will not have the subworkflowmetdata key and will fail when it approaches the "for" loop.

Reply:
great, we've cleared one confusion we have!